multisense_ros/src/reconfigure.cpp

```diff
@@ -78,6 +80,20 @@ Reconfigure::Reconfigure(Channel* driver,
         return;
     }
 
+    status = driver_->getDeviceModes(deviceModes);
+    if (Status_Ok != status) {
+        ROS_ERROR("Reconfigure: failed to query device modes: %s",
+                  Channel::statusString(status));
+        return;
+    }
+
+    bool ground_surface_supported = false;
+    for (auto &mode : deviceModes)
+    {
+        ground_surface_supported |= (mode.supportedDataSources & Source_Ground_Surface_Spline_Data) &&
+                                    (mode.supportedDataSources & Source_Ground_Surface_Class_Image);
+    }
```

Comment:
Here's where we check whether the ground surface streams are available on the camera

Reply:
You are using the bitwise or assignment operator. You should modify to be:

```
ground_surface_supported = (mode.supportedDataSources & Source_Ground_Surface_Spline_Data) &&                                  (mode.supportedDataSources & Source_Ground_Surface_Class_Image);
```

Reply:
Nice spot, thanks for that — must've been a late night change